src/System.CommandLine/Option.cs

```diff
@@ -170,11 +171,17 @@ public virtual IArgumentArity Arity
                 {
                     Argument.ValueType = typeof(string);
                 }
-                
+
                 Argument.Arity = value;
             }
         }
 
+        /// <summary>
+        /// Global options are applied to the command and recursively to subcommands.
+        /// They do not apply to parent commands.
+        /// </summary>
+        public bool IsGlobal { get; internal set; }
```

Comment:
Can we make this internal?

Reply:
What was the reason for this change? Does this improve performance and if yes how/why? There is no way (from outside) to differentiate between global and local options anymore.

Reply:
@lennybacon they were duplicated (added to two collections). This commit includes all the changes related to it: https://github.com/dotnet/command-line-api/pull/1510/commits/d38bae9846bdf87039cb0b846a55ff15685510d4

> There is no way (from outside) to differentiate between global and local options anymore.

Ech option exposed by `.Options` is not providing `IsGlobal` property

Reply:
@adamsitnik I got that. 

> Ech option exposed by .Options is not providing IsGlobal property

Not publicly, but the property is still there - it is internal.

I used this and now can only use reflection to access this information.

**My question was to understand weather a property being public vs. internal improves performance.**

And if performance was not the reason for the change... what was ist?

Reply:
@lennybacon please excuse me, I missed the fact that the getter is not public anymore.

> My question was to understand weather a property being public vs. internal improves performance.

it does not affect performance

> And if performance was not the reason for the change... what was ist?

most likely encapsulation, @jonsequitur do you remember why we have made `IsGlobal` an `internal` property?

Reply:
Thank you @adamsitnik 

To explain what I was doing: I created an *interactive* global option. That when provided reads the command line and jumps in at any stage to allow interactive selection of command/argument/option which results in a newly created command line that is then displayed and in the end also executed. I really enjoyed the *explorable* style of System.CommandLine and would love to continue using it without dirty hacks like reflection.

Reply:
@jonsequitur would you like me to create an issue (feature request) to make `IsGlobal` public again?

Reply:
> @jonsequitur would you like me to create an issue (feature request) to make IsGlobal public again?

That would be great. I'd recommend a more problem-oriented approach though. What's the thing you're trying to accomplish? Given this bug, making `IsGlobal` public might not be the only way to accomplish it, and we can use the issue to discuss approaches.

Reply:
I created an issue here: https://github.com/dotnet/command-line-api/issues/1736